libs/application/template-api-modules/src/lib/modules/shared/api/national-registry/national-registry.service.ts

```diff
@@ -31,39 +30,100 @@ export class NationalRegistryService extends BaseTemplateApiService {
     auth,
     params,
   }: TemplateApiModuleActionProps<NationalRegistryParameters>): Promise<NationalRegistryIndividual | null> {
-    const result = await this.getIndividual(auth.nationalId)
-    // Make sure user has domicile country as Iceland
-    if (params?.legalDomicileIceland) {
-      const domicileCode = result?.address?.municipalityCode
-      if (!domicileCode || domicileCode.substring(0, 2) === '99') {
-        throw new TemplateApiError(
-          {
-            title: coreErrorMessages.nationalRegistryLegalDomicileNotIceland,
-            summary: coreErrorMessages.nationalRegistryLegalDomicileNotIceland,
-          },
-          400,
-        )
-      }
+    const individual = await this.getIndividual(auth.nationalId)
+
+    //Check if individual is found in national registry
+    if (!individual) {
+      throw new TemplateApiError(
+        {
+          title: coreErrorMessages.nationalIdNotFoundInNationalRegistryTitle,
+          summary:
+            coreErrorMessages.nationalIdNotFoundInNationalRegistrySummary,
+        },
+        404,
+      )
+    }
+
+    // Case when parent can apply for custody child without fulfilling some requirements
+    if (params?.allowPassOnChild) {
+      const children = await this.nationalRegistryApi.getCustodyChildren(auth)
+      this.validateChildren(params, children)
+    }
+
+    // Validate individual
+    this.validateIndividual(individual, false, params)
+
+    return individual
+  }
+
+  private validateIndividual(
+    individual: NationalRegistryIndividual,
+    isChild: boolean,
+    params?: NationalRegistryParameters,
+  ) {
+    if (params?.legalDomicileIceland && !params?.allowPassOnChild) {
+      this.validateDomicleInIceland(individual, params.allowPassOnChild)
```

Comment:
Won't `params.allowPassOnChild` always be `false` or `undefined` here?
Is that expected?

Reply:
your right!